nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/SplitText.java

```diff
@@ -150,548 +145,320 @@
-                                    final long procNanos = System.nanoTime() - startNanos;
-                                    final long procMillis = TimeUnit.MILLISECONDS.convert(procNanos, TimeUnit.NANOSECONDS);
-                                    logger.debug("Detected start of Split File in {} at byte offset {} with a length of {} bytes; "
-                                                    + "total splits = {}; total processing time = {} ms",
-                                            new Object[]{flowFile, beforeReadingLines, info.lengthBytes, splitInfos.size(), procMillis});
-                                }
+                                splitFlowFile = session.clone(flowFile, computedSplitInfo.startOffset, length);
                             }
+
+                            splitFlowFile = SplitText.this.updateAttributes(session, splitFlowFile, computedSplitInfo.lineCount, splitFlowFile.getSize(), fragmentId, fragmentIndex++,
+                                    computedSplitsInfo.size(), flowFile.getAttribute(CoreAttributes.FILENAME.key()));
+                            splitFlowFiles.add(splitFlowFile);
                         }
                     }
                 }
-            }
-        });
 
-        if (errorMessage.get() != null) {
-            logger.error("Unable to split {} due to {}; routing to failure", new Object[]{flowFile, errorMessage.get()});
-            session.transfer(flowFile, REL_FAILURE);
-            if (!splits.isEmpty()) {
-                session.remove(splits);
+                getLogger().info("Split " + flowFile + " into " + splitFlowFiles.size() + " flow files" + (headerFlowFile != null ? " containing headers." : "."));
+                if (headerFlowFile != null) {
+                    session.remove(headerFlowFile);
+                }
             }
-            return;
-        }
 
-        if (!splitInfos.isEmpty()) {
-            // Create the splits
-            for (final SplitInfo info : splitInfos) {
-                FlowFile split = session.clone(flowFile, info.offsetBytes, info.lengthBytes);
-                split = session.putAttribute(split, SPLIT_LINE_COUNT, String.valueOf(info.lengthLines));
-                split = session.putAttribute(split, FRAGMENT_SIZE, String.valueOf(info.lengthBytes));
-                splits.add(split);
+            if (error.get()) {
+                session.transfer(flowFile, REL_FAILURE);
+            } else {
+                session.transfer(flowFile, REL_ORIGINAL);
+                session.transfer(splitFlowFiles, REL_SPLITS);
             }
-        }
-        finishFragmentAttributes(session, flowFile, splits);
-
-        if (splits.size() > 10) {
-            logger.info("Split {} into {} files", new Object[]{flowFile, splits.size()});
         } else {
-            logger.info("Split {} into {} files: {}", new Object[]{flowFile, splits.size(), splits});
+            context.yield();
```

Comment:
We should not be yielding in this case. There is nothing preventing the processor from making progress on the next FlowFile.